packages/core/src/execution-engine/node-execution-context/utils/get-input-connection-data.ts

```diff
@@ -28,19 +28,28 @@ import type { ExecuteContext, WebhookContext } from '../../node-execution-contex
 // eslint-disable-next-line import/no-cycle
 import { SupplyDataContext } from '../../node-execution-context/supply-data-context';
 
+function getNextRunIndex(runExecutionData: IRunExecutionData, nodeName: string) {
```

Comment:
can we avoid this duplication and use `getNextRunIndex` from the context instead?

Reply:
I am not sure how, it looks to me like a chicken-and-the-egg problem :)
I tried to do something like 
```js
	const tempContext = new SupplyDataContext(
		{} as Workflow,
		node,
		{} as IWorkflowExecuteAdditionalData,
		'internal' as WorkflowExecuteMode,
		runExecutionData,
		0,
		[] as INodeExecutionData[],
		{} as ITaskDataConnections,
		NodeConnectionTypes.AiTool,
		{} as IExecuteData,
		[] as CloseFunction[],
	);
	let runIndex = tempContext.getNextRunIndex();
```
but that just looks sad :( 

Happy to remove the function (it's really only called in here) and do 
```js
let runIndex = runExecutionData.resultData.runData[node.name]?.length ?? 0;
```

but is there another way?